monitoring/rid_qualifier/test_executor.py

```diff
@@ -40,3 +56,9 @@ def main(test_configuration: RIDQualifierTestConfiguration, auth_spec: str):
         report.findings)
     with open('report.json', 'w') as f:
         json.dump(report, f)
+    if job_id:
```

Comment:
Instead of adding extra behaviors to `main`, let's split out most of the functionality originally in `main` into something like `conduct_test` which only returns `report` (and doesn't write it to file).  Then, `main` can call `conduct_test` and write the result to file while the Redis worker can do something else with the report.  Importantly, `rid_qualifier` by itself should probably not need to do anything with Redis (currently, `main` is trying to get the current Redis job) since it's just the `host` which is employing a Redis worker to run it.